Makefile

```diff
@@ -1,10 +1,82 @@
 SHELL := /bin/bash
+PYTHON ?= python
+PYTHON3 ?= python3
 
-dockerized_tests:
-	python ./scripts/start_services.py
+COMPOSE_ARGS ?=
 
-tests:
-	python ./scripts/wait_until_services_running.py ${URLS}
-	${TEST_CMD}
+# Make sure we run local versions of everything, particularly commands
+# installed into our virtualenv with pip eg. `docker-compose`.
+export PATH := ./bin:./venv/bin:$(PATH)
 
-.PHONY: dockerized_tests tests
+all: test
+
+# The tests are written in Python. Make a virtualenv to handle the dependencies.
+venv: requirements.txt
+	test -d venv || virtualenv --python=$(PYTHON3) venv;\
+	pip install -r requirements.txt;\
+	touch venv;\
+
+lint: venv
+	flake8 tests/
+.PHONY: lint
+
+start-env: venv
+	$(PYTHON) scripts/compose.py start $(COMPOSE_ARGS)
+	docker-compose up -d
+
+stop-env: venv
+	docker-compose down -v --remove-orphans || true
+
+destroy-env: venv
+	(docker ps -aqf network=apm-integration-testing | xargs docker rm -f && docker network rm apm-integration-testing) || true
+
+# default (all) started for now
+env-%: venv
+	$(MAKE) start-env
+
+test: test-all lint
```

Comment:
would be nice to document briefly all these targets: what do they do, what depends on what, if i am meant to run something manually or they are only used internally, etc.